python/lsst/pipe/tasks/processImage.py

```diff
@@ -100,25 +92,12 @@ class ProcessImageTask(pipeBase.CmdLineTask):
     ConfigClass = ProcessImageConfig
     dataPrefix = ""  # Name to prepend to all input and output datasets (e.g. 'goodSeeingCoadd_')
 
-    def __init__(self, **kwargs):
+    def __init__(self, schema=None, **kwargs):
         pipeBase.CmdLineTask.__init__(self, **kwargs)
-
-        self.makeSubtask("calibrate")
-
-        # Setup our schema by starting with fields we want to propagate from icSrc.
-        calibSchema = self.calibrate.schema
-        self.schemaMapper = afwTable.SchemaMapper(calibSchema)
-        self.schemaMapper.addMinimalSchema(afwTable.SourceTable.makeMinimalSchema(), False)
-
-        # Add fields needed to identify stars used in the calibration step
-        self.calibSourceKey = self.schemaMapper.addOutputField(
-            afwTable.Field["Flag"]("calib_detected", "Source was detected as an icSrc")
-            )
-
-        for key in self.calibrate.getCalibKeys():
-            self.schemaMapper.addMapping(key)
-        self.schema = self.schemaMapper.getOutputSchema()
         self.algMetadata = dafBase.PropertyList()
+        if schema is None:
+            schema = afwTable.SourceTable.makeMinimalSchema()
+        self.schema = schema
```

Comment:
Same question as earlier -- I don't object to the schema dance per se, but I don't see why it's folded into this commit.